src/commands/build-image.yml

```diff
@@ -39,20 +39,56 @@ parameters:
       Extra flags to pass to docker build. For examples, see
       https://docs.docker.com/engine/reference/commandline/build
 
+  skip-build-when-tags-exist-in-ecr:
+    type: boolean
+    default: false
+    description: Whether to skip image building if all specified tags already exist in ECR
+
+  region:
+    type: env_var_name
+    default: AWS_REGION
+    description: >
+      Name of env var storing your AWS region information,
+      defaults to AWS_REGION. Only required when skip-build-when-tags-exist-in-ecr
+      is set to true.
+
+  profile-name:
+    type: string
+    default: "default"
+    description: >
+      AWS profile name to be configured. Only required when skip-build-when-tags-exist-in-ecr
+      is set to true.
+
 steps:
+  - when:
+      condition: <<parameters.skip-build-when-tags-exist-in-ecr>>
+      steps:
+        - ecr-login:
+            profile-name: <<parameters.profile-name>>
+            region: <<parameters.region>>
+            account-url: <<parameters.account-url>>
+
   - run:
       name: Build docker image
       command: |
+        registry_id=$(echo <<parameters.account-url>> | sed "s;\.\w;;g")
+        docker_tags_exist_in_ecr="false"
+
         docker_tag_args=""
         IFS="," read -ra DOCKER_TAGS \<<< "<< parameters.tag >>"
         for tag in "${DOCKER_TAGS[@]}"; do
+          if [ "<<parameters.skip-build-when-tags-exist-in-ecr>>" = "true" ]; then
+            docker_tags_exist_in_ecr=$(aws ecr describe-images --registry-id $registry_id --repository-name <<parameters.repo>> --query "contains(imageDetails[].imageTags[], '$tag')")
+          fi
           docker_tag_args="$docker_tag_args -t $<<parameters.account-url>>/<<parameters.repo>>:$tag"
         done
-        docker build \
-          <<#parameters.extra-build-args>><<parameters.extra-build-args>><</parameters.extra-build-args>> \
-          -f <<parameters.path>>/<<parameters.dockerfile>> \
-          $docker_tag_args \
-          <<parameters.path>>
+        if [ "<<parameters.skip-build-when-tags-exist-in-ecr>>" = "false" ] || [ "<<parameters.skip-build-when-tags-exist-in-ecr>>" = "true" -a "$docker_tags_exist_in_ecr" = "false" ]; then
+          docker build \
+            <<#parameters.extra-build-args>><<parameters.extra-build-args>><</parameters.extra-build-args>> \
+            -f <<parameters.path>>/<<parameters.dockerfile>> \
+            $docker_tag_args \
+            <<parameters.path>>
+        fi
```

Comment:
Can you confirm this logic has been tested locally? I'll be trying to confirm this myself but don't have a testing ground yet.

Reply:
I thought the tests in the circleci config would be running all the time? That being said, I didn't see any feedback coming back to the PR

Reply:
hi @gmemstr I've seen that you triggered the job in circleci - could you please retrigger https://app.circleci.com/pipelines/github/CircleCI-Public/aws-ecr-orb/276/workflows/ef4cfb8c-40b6-40db-bfe2-12c47f5fcb10 as I've pushed out a fix for the failure?

Reply:
and I had another issue, with the regex this time... any chance you could retrigger it please?

Reply:
sorry to be a pain but I've now pushed another commit to reorder the create repo step - any chance the correct workflow could be automatically run on every push? :)

Reply:
sorry, couple more things fixed :)

Reply:
and hopefully last one - tests need to be sequential as otherwise they delete each other's ecr repo

Reply:
finally - all green now :)